test_hstack.py

```
@@ -0,0 +1,231 @@
+"""
+Test script to validate our custom hstack function
+works similarly to scipy's sp.hstack.
+"""
+
+def test_hstack_comparison():
+    """Compare our hstack function with scipy's sp.hstack behavior"""
+    
+    # Import scipy for comparison (we'll remove this later)
+    try:
+        import scipy.sparse as sp
+        import numpy as np
```

Ruff (0.11.9)

`numpy` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)